### PR TITLE
Moving Slice to Buffers Experimental and using AsSpan instead

### PR DESCRIPTION
**Pull request by ahsonkhan**
**WIP (CI will fail for now).**

**As prep for:**
https://github.com/dotnet/corefxlab/issues/1314 / https://github.com/dotnet/corefxlab/issues/1292

**This won't pass until AsSpan() becomes available, i.e. this PR goes through:**
https://github.com/dotnet/corefx/pull/17407

Waiting for these additions:
```C#
    public static ReadOnlySpan<char> AsSpan(this string text);
    public static Span<T> AsSpan<T>(this T[] array);
    public static Span<T> AsSpan<T>(this ArraySegment<T> segment);
```

Cleaning up csproj files of projects that don't use/need Buffers.Experimental.csproj.

### CODE REVIEW

File: src/System.Buffers.Primitives/System/SpanExtensions_array.cs [hunk 1]
```diff
@@ -12,67 +12,9 @@ namespace System
...
-        /// </exception>
-        public static Span<T> Slice<T>(this T[] array, int start, int length)
-        {
-            return new Span<T>(array, start, length);
-        }
-
         public static void CopyTo<T>(this T[] array, Span<T> span)
         {
-            array.Slice().CopyTo(span);
+            new Span<T>(array).CopyTo(span);
```
Comment (davidfowl) [inline src/System.Buffers.Primitives/System/SpanExtensions_array.cs]: As Span?

**Comment by ahsonkhan** [comment]
@dotnet-bot test this please

**Comment by ahsonkhan** [comment]
Once [this] (https://github.com/dotnet/corefxlab/pull/1357) PR is merged, CI should go green here.

**Comment by ahsonkhan** [comment]
While we wait for this PR: https://github.com/dotnet/corefx/pull/17744, I have added a work around to get corefxlab building (for now).

**Pull request merged**